topdown/http.go

```diff
@@ -253,8 +275,10 @@ func executeHTTPRequest(bctx BuiltinContext, obj ast.Object) (ast.Value, error)
 	}
 
 	// check if redirects are enabled
-	if enableRedirect {
-		client.CheckRedirect = nil
+	if !enableRedirect {
+		client.CheckRedirect = func(*http.Request, []*http.Request) error {
+			return http.ErrUseLastResponse
+		}
```

Comment:
Is this an important fix? What was happening before? Maybe a comment would be useful for our future selves.

Reply:
Nothing changed with the behavior, its just a refactor with how the client is being created. It used to set this explicitly on the client creation and then `nil` it out if redirects are enabled. Now we don't set it when creating the client and just update it as needed (when redirects are disabled).